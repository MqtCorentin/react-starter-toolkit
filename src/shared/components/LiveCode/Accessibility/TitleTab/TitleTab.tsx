import { Result } from 'axe-core';

const TitleTab = ({ results, label, icon }: { results: Result[] | undefined; label: string; icon: string }) => (
  <>
    <i className={`glyphicon glyphicon-${icon}`} /> ({results?.length || 0}) {label}
  </>
);

export default TitleTab;
