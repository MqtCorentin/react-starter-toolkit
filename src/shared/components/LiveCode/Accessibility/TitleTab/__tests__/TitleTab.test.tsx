import { render } from '@testing-library/react';
import TitleTab from '../TitleTab';

const resultsFnMock = [
  {
    description: 'test',
    help: 'test',
    helpUrl: 'url',
    impact: 'critical',
    tags: ['tag1'],
  },
];
describe('<TitleTab />', () => {
  it('Render <AccordionResults /> with results, label, icon', () => {
    const { asFragment } = render(<TitleTab results={resultsFnMock} label="test" icon="ok" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
