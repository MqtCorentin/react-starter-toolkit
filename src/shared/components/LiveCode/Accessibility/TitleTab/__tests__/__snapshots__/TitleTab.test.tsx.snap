// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleTab /> Render <AccordionResults /> with results, label, icon 1`] = `
<DocumentFragment>
  <i
    class="glyphicon glyphicon-ok"
  />
   (1) test
</DocumentFragment>
`;
